charts/redpanda/templates/post-install-upgrade-job.yaml

```diff
@@ -76,23 +76,27 @@ spec:
 {{- if .Values.auth.sasl.enabled }}
   {{- $values := .Values }}
   {{- range $user := .Values.auth.sasl.users }}
-            # To avoid `set -e` from exiting the command when a user exists; catch the stderr output and exit codes into `creation_result`
-            # and `creation_result_exit_code` for use later
-            creation_result=$(rpk acl user create {{ $user.name }} -p {{ $user.password | quote }} --mechanism {{ include "sasl-user-mechanism" (dict "user" $user "Values" $values) }} {{ template "rpk-common-flags" $ }} 2>&1) && creation_result_exit_code=$? || creation_result_exit_code=$?
-
-            # On a non-success exit code
+    {{- if $user.password }}
+            PASSWORD={{ $user.password | quote }}
+    {{- else if not (empty $user.secretName) }}
+            PASSWORD=$(cat {{ printf "/etc/secret-name/%s/password" $user.secretName }})
+    {{- else }}
+            PASSWORD=""
```

Comment:
If user does not provide `secretName` nor `password` could we add Notes for them to set the password. I don't even now if Redpanda Admin API allows service account creation without any password.

Reply:
There is already a validation check. I left the default password there uncommented because we should use defaults. The validation error is 
```
Error: INSTALLATION FAILED: values don't meet the specifications of the schema(s) in the following chart(s):
redpanda:
- auth.sasl: Must validate "then" as "if" was valid
- auth.sasl.users.0: Must validate one and only one schema (oneOf)
- auth.sasl.users.0: password is required

```

Reply:
What was the values provided? `--set auth.sasl.enable=true` or more?

Reply:
```
--set auth.sasl.enable=true
```

Yes, i uncommented the password for that particular case so as to avoid this issue, but now reverted it as i think having a user manually set their passwords would be more appropriate. Personally I would prefer to avoid "creating" passwords for users.